src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1576,20 +1626,41 @@ public static BigInteger Pow(BigInteger value, Int32 exponent)
             left.AssertValid();
             right.AssertValid();
 
-            if (right.IsZero) return left;
-            if (left.IsZero) return right;
+            if (right._sign == 0)
+                return left;
+            if (left._sign < 0 != right._sign < 0)
+                return left - (-right);
 
-            int sign1 = +1;
-            int sign2 = +1;
-            BigIntegerBuilder reg1 = new BigIntegerBuilder(left, ref sign1);
-            BigIntegerBuilder reg2 = new BigIntegerBuilder(right, ref sign2);
+            bool trivialLeft = left._bits == null;
+            bool trivialRight = right._bits == null;
 
-            if (sign1 == sign2)
-                reg1.Add(ref reg2);
-            else
-                reg1.Sub(ref sign1, ref reg2);
+            if (trivialLeft && trivialRight)
+            {
+                return (long)left._sign + right._sign;
+            }
 
-            return reg1.GetInteger(sign1);
+            if (trivialLeft)
+            {
+                uint[] bits = BigIntegerCalculator.Add(right._bits, NumericsHelpers.Abs(left._sign));
+                return new BigInteger(bits, left._sign < 0);
+            }
+
+            if (trivialRight)
+            {
+                uint[] bits = BigIntegerCalculator.Add(left._bits, NumericsHelpers.Abs(right._sign));
+                return new BigInteger(bits, left._sign < 0);
+            }
+
+            if (left._bits.Length < right._bits.Length)
```

Comment:
Why is it better to do this check here and effectively duplicate the code in the blocks (and anywhere else the Add function might get called) rather than moving this check into the Add function such that it can work regardless of whether left._bits.Length < or > than right._bits.Length?